### PR TITLE
Add SVG topology to bundles

### PR DESCRIPTION
**Pull request by solazio**
## Done
- _Add SVG topology to bundles oveview page._

## How to QA
- Visit the demo in your browser, link commented below
  - Or you can run the project locally using the [dotrun](https://snapcraft.io/dotrun) snap with `$ dotrun` and view it in your web browser at: http://localhost:8045/kubeflow
  - If your demo doesn't work, read more about the [demoservice](https://discourse.ubuntu.com/t/demoservice/16862)
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Compare to [design 1](https://app.zeplin.io/project/5f92a9530029eeac84ae9589/screen/5fc0d993e26203b80414cf29) and [design 2](https://app.zeplin.io/project/5f92a9530029eeac84ae9589/screen/600024735687ea92014a2ada)

## Issue / Card
Fixes #

### CODE REVIEW

**Review by anthonydillon — COMMENTED** [review]
Why not just add the topology to the template above the readme? It will give you more freedom to mark it up and its where the next developer will first look to find it.

**Comment by solazio** [comment]
> Why not just add the topology to the template above the readme? It will give you more freedom to mark it up and its where the next developer will first look to find it.
> 
> Does the design ask for it to be injected into the readme?

@danielmutis said that he wants the title to come first as per the [design](https://app.zeplin.io/project/5f92a9530029eeac84ae9589/screen/5fc0d993e26203b80414cf29). @danielmutis what do you think?

**Comment by tbille** [comment]
The main title of the page is the bundle name on the header.

As  a developer I would expect to see my readme not modified with injected image. It seems more natural to have before the readme to get a quick understanding of the bundle.

It also removes a code complexity, and we like simple code :laughing:

**Comment by danielmutis** [comment]
@solazio Let's push the whole readme below the image then.

**Comment by anthonydillon** [comment]
You could put the topology in its own tab?

File: static/js/src/public/details/index.js [hunk 1]
```diff
@@ -31,8 +31,37 @@ const truncateSummary = (selector) => {
...
+    const delta = 300;
+    const width = vw - delta;
+    const height = vh - delta;
+    const closeModalButton = topologyModal.querySelector(
+      "[aria-controls='modal']"
+    );
+    const topologyImage = topologyModal.querySelector("object");
+
+    topologyImage.style.width = `${width}px`;
+    topologyImage.style.height = `${height}px`;
```
Comment (anthonydillon) [inline static/js/src/public/details/index.js:54]: Why not use CSS to style and position the topology object? You will need to handle resizes if you are styling with us.

File: templates/details/overview.html [hunk 2]
```diff
@@ -13,6 +13,29 @@ <h4 class="p-heading--5 u-no-margin--bottom">About</h4>
...
         <hr class="p-separator--shallow">
       {% endif %}
+      {% if package.type == "bundle" %}
+        {% if not package.result.summary %}
+          <h4 class="p-heading--5 u-sv1 u-no-margin--bottom">About</h4>
+        {% endif %}
+          <div class="p-topology-info" data-js="topology-info">
+            <div class="p-card--highlighted">
+              <object data="https://api.jujucharms.com/charmstore/v5/bundle/{{ package.name }}/diagram.svg" type="image/svg+xml" style="pointer-events: none;"></object>
+              <i class="p-icon--fullscreen" role="button" tabIndex="0">Show fullscreen</i>
```
Comment (anthonydillon) [inline templates/details/overview.html:23]: This is not keyboard accessible can you make this a base button and attach the open click event listener to it please?

File: static/js/src/public/details/index.js [hunk 3]
```diff
@@ -31,8 +31,37 @@ const truncateSummary = (selector) => {
...
+
+    topologyInfoPanel.addEventListener("click", (e) => {
+      e.preventDefault();
+      topologyModal.classList.remove("u-hide");
+    });
+
+    closeModalButton.addEventListener("click", (e) => {
```
Comment (anthonydillon) [inline static/js/src/public/details/index.js:61]: ESC should also close this modal.

File: templates/details/overview.html [hunk 4]
```diff
@@ -13,6 +13,29 @@ <h4 class="p-heading--5 u-no-margin--bottom">About</h4>
...
+            </div>
+          </div>
+          <div class="p-modal u-hide" data-js="topology-modal">
+            <div class="p-modal__dialog" role="dialog" aria-labelledby="modal-title" aria-describedby="modal-description">
+              <header class="p-modal__header">
+                <h2 class="p-modal__title" id="modal-title">{{ package.name }}</h2>
+                <button class="p-modal__close" aria-label="Close active modal" aria-controls="modal">Close</button>
+              </header>
+              <object data="https://api.jujucharms.com/charmstore/v5/bundle/{{ package.name }}/diagram.svg" type="image/svg+xml"></object>
```
Comment (anthonydillon) [inline templates/details/overview.html:32]: Can you add `id="modal-description"` to complete the linking of the data.

File: templates/details/overview.html [hunk 5]
```diff
@@ -13,6 +13,29 @@ <h4 class="p-heading--5 u-no-margin--bottom">About</h4>
...
+            <div class="p-card--highlighted">
+              <object data="https://api.jujucharms.com/charmstore/v5/bundle/{{ package.name }}/diagram.svg" type="image/svg+xml" style="pointer-events: none;"></object>
+              <i class="p-icon--fullscreen" role="button" tabIndex="0">Show fullscreen</i>
+            </div>
+          </div>
+          <div class="p-modal u-hide" data-js="topology-modal">
+            <div class="p-modal__dialog" role="dialog" aria-labelledby="modal-title" aria-describedby="modal-description">
+              <header class="p-modal__header">
+                <h2 class="p-modal__title" id="modal-title">{{ package.name }}</h2>
+                <button class="p-modal__close" aria-label="Close active modal" aria-controls="modal">Close</button>
```
Comment (anthonydillon) [inline templates/details/overview.html:30]: label could be more descriptive. For example, "Close the topology modal"

Comment (anthonydillon) [reply templates/details/overview.html:30]: There is no `id="model"` which is declared in the `aria-controls` value. I think you mean to add the id to the model container?

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Thanks @solazio, few comments inline.

File: static/js/src/public/details/index.js (hunk 1 above)
Comment (solazio) [reply static/js/src/public/details/index.js:54]: Agree. thanks

**Review by anthonydillon — APPROVED** [review]
LGTM :+1:

**Pull request merged**